### DISCUSSION TITLE
Installation Error when installing "OpenSwoole" from code as header file <ares.h> not found

### DISCUSSION THREAD

**Discussion by fakharak** [swoole/swoole-src#4491] [Q&A] [answered]
OS: Ubuntu 18.04
PHP version 8.0
Installation Command:
cd /tmp && git clone https://github.com/openswoole/swoole-src.git && \
    cd swoole-src && \
    git checkout v4.7.2 && \
    phpize8.0  && \
    ./configure --enable-openssl \
                --enable-mysqlnd \
                --enable-sockets \
                --enable-http2 \
                --enable-swoole-curl \
                --enable-swoole-json \
                --enable-debug \
                --enable-debug-log \
                --enable-trace-log \
                --enable-thread-context \
                --enable-cares \
                --with-php-config=/usr/bin/php-config8.0 && \
    sudo make && sudo make install
    
    ---------------
    Error Message: /tmp/swoole-src/ext-src/php_swoole.cc:54:10: fatal error: ares.h: No such file or directory
-------------------

Before installing Swoole, i installed c-ares library on ubuntu using command below:
sudo apt-get install -y libc-ares2
![swoole_installation_issue](https://user-images.githubusercontent.com/9947110/141655332-bc47aad7-c7cc-496c-89d7-ea8733bb319f.png)

**Comment by fakharak** [comment]
The issue resolved by installing "libc-ares-dev", then removed the "swoole-src" folder, and re-installing openswoole from source again (starting from cloning git repo), so pre-requisite to configure compile-time swoole option "--enable-cares" is:
sudo apt-get install -y libc-ares-dev

**Comment by matyhtf** [comment, marked as answer]
You added `--enable-cares` to the compilation configuration, tha requires `libc-ares-dev` to be installed
